app/models/attendance.rb

```diff
@@ -32,23 +34,94 @@ class Attendance < UuidApplicationRecord
   }
 
+  scope :absences, -> { where.not(absence: nil) }
+  scope :non_absences, -> { where(absence: nil) }
+
   scope :illinois_part_days, -> { where('total_time_in_care < ?', '5 hours') }
   scope :illinois_full_days, -> { where('total_time_in_care BETWEEN ? AND ?', '5 hours', '12 hours') }
   scope :illinois_full_plus_part_days, -> { where('total_time_in_care > ? AND total_time_in_care < ?', '12 hours', '17 hours') }
   scope :illinois_full_plus_full_days, -> { where('total_time_in_care BETWEEN ? AND ?', '17 hours', '24 hours') }
 
+  delegate :business, to: :child_approval
+  delegate :user, to: :child_approval
+  delegate :child, to: :child_approval
+  delegate :state, to: :child
+  delegate :county, to: :child
+  delegate :timezone, to: :user
+
   private
 
+  def round_check_in
+    return unless check_in
+
+    self.check_in = Time.at(check_in - check_in.sec).in_time_zone(timezone)
+  end
+
+  def round_check_out
+    return unless check_out
+
+    self.check_out = Time.at(check_out - check_out.sec).in_time_zone(timezone)
+  end
+
   def calc_total_time_in_care
     self.total_time_in_care = if check_in && check_out
-                                check_out - check_in
-                              elsif child_approval.child.state == 'NE'
+                                (check_out - check_in).round(2)
+                              elsif state == 'NE'
                                 calculate_from_schedule
                               else
                                 0.seconds
                               end
   end
 
+  def calc_earned_revenue
+    return unless state == 'NE'
+
+    self.earned_revenue = child_approval.special_needs_rate ? ne_special_needs_revenue : ne_base_revenue
+  end
+
+  def ne_hours
+    # TODO: this is super sloppy because this shouldn't be a service class but we haven't refactored these to procedures yet
+    NebraskaHoursCalculator.new(child, check_in).round_hourly_to_quarters(total_time_in_care.seconds)
+  end
+
+  def ne_days
+    # TODO: this is super sloppy because this shouldn't be a service class but we haven't refactored these to procedures yet
+    NebraskaFullDaysCalculator.new(child, check_in).calculate_full_days_based_on_duration(total_time_in_care.seconds)
+  end
+
+  # TODO: open question - does qris bump impact this rate?
+  def ne_special_needs_revenue
+    (ne_hours * child_approval.special_needs_hourly_rate) + (ne_days * child_approval.special_needs_daily_rate)
+  end
+
+  def ne_base_revenue
+    (ne_hours * ne_hourly_rate * business.ne_qris_bump) + (ne_days * ne_daily_rate * business.ne_qris_bump)
+  end
+
+  def ne_hourly_rate
+    # TODO: License Types - possibly post-new-data-model
+    ne_rates.hourly.first&.amount || 0
+  end
+
+  def ne_daily_rate
+    # TODO: License Types - possibly post-new-data-model
+    ne_rates.daily.first&.amount || 0
+  end
+
+  def ne_rates
+    NebraskaRate
```

Comment:
This is where we find all the rates that could feasibly apply to this child, based on age, whether or not they're enrolled in school, the business's accreditation, the region, etc.  `order_max_age` is a scope on Nebraska rates that sorts ascending by max_age and forces the nulls to the end.  This way if we have a child who is 2, a rate that will apply up to 36 months, and a rate that will apply after 36 months with *no* upper limit on max age, the child will get the 36 month one instead.